llvm/lib/IR/DataLayout.cpp

```diff
@@ -545,6 +545,7 @@ void DataLayout::init(const Module *M) { *this = M->getDataLayout(); }
 
 bool DataLayout::operator==(const DataLayout &Other) const {
   bool Ret = BigEndian == Other.BigEndian &&
+             HasCheriCapabilities == Other.HasCheriCapabilities &&
```

Comment:
Yes, it should be. Does keeping it here cause any harm? I think it's easier to just compare all fields so that someone looking at this later doesn't wonder whether omitting it was intentional or not. I guess we could also add a comment if you'd prefer that.